app/controllers/multifactor_auths_controller.rb

```diff
@@ -1,8 +1,13 @@
 class MultifactorAuthsController < ApplicationController
+  include MfaExpiryMethods
+  include WebauthnVerifiable
+
   before_action :redirect_to_signin, unless: :signed_in?
-  before_action :require_mfa_disabled, only: %i[new create]
-  before_action :require_mfa_enabled, only: :update
+  before_action :require_totp_disabled, only: %i[new create]
+  before_action :require_mfa_enabled, only: %i[update destroy]
```

Comment:
Wouldn't `destroy` need to be in `require_totp_enabled`? `require_mfa_enabled` wouldn't work as mfa is enabled if someone has a webauthn device and not totp.